src/components/TimeSelectInput/helpers/__test__/normalizeValue.spec.js

```diff
@@ -0,0 +1,137 @@
+import normalizeValue from '../normalizeValue';
+
+jest.mock('../getHour', () => value => {
+    if (value) {
+        return value.split(':')[0];
+    }
+    return '';
+});
+
+jest.mock('../getMinutes', () => value => {
+    if (value) {
+        return value.split(':')[1].split(' ')[0];
+    }
+    return '';
+});
+
+jest.mock('../getAmPm', () => value => {
+    if (value) {
+        return value.split(' ')[1];
+    }
+    return undefined;
+});
+
+jest.mock('../getAmPmValue', () => value => {
+    return +value > 11 ? 'PM' : 'AM';
+});
+
+jest.mock('../get24Hour', () => (hour, ampm) => {
+    if (!ampm) {
+        return hour;
+    }
+    if (ampm === 'AM') {
+        if (+hour === 12) {
+            return '00';
+        }
+        return hour;
+    }
+    if (ampm === 'PM' && +hour < 13) {
+        if (+hour === 12) {
+            return '12';
+        }
+        return String(+hour + 12);
+    }
+    return hour;
+});
+
+jest.mock('../get12Hour', () => hour => {
+    if (+hour === 0) {
+        return '12';
+    }
+    if (+hour > 12) {
+        if (+hour < 22) {
+            return `0${String(+hour - 12)}`;
+        }
+        return String(+hour - 12);
+    }
+    return hour;
+});
+
+jest.mock('../normalizeHour', () => (value, hour24) => {
+    if (+value === 0 && !hour24) {
+        return '12';
+    }
+    if (+value < 10) {
+        return `0${+value}`;
+    }
+    if (+value > 12 && +value < 20 && !hour24) {
+        return `0${+value - 12}`;
+    }
+    return String(+value);
+});
```

Comment:
we should not reimplement helpers in its mocks, since it is difficult to read and understand, also how you guaranties the right value returned, if we use this complex implementations in the mocks the we would need to test the mock itself, instead what you can do is to return a simple value like an empty string and then in each test where a specific value is needed, you can do:
```normalizeHour.mockReturnValue('12')```